src/Compilers/Core/Portable/Compilation/Compilation.cs

```diff
@@ -824,6 +824,51 @@ public INamedTypeSymbol GetSpecialType(SpecialType specialType)
         protected abstract INamedTypeSymbol CommonScriptClass { get; }
 
         /// <summary>
+        /// Resolves a symbol that represents script container (Script class). Uses the
+        /// full name of the container class stored in <see cref="CompilationOptions.ScriptClassName"/> to find the symbol.
+        /// </summary>
+        /// <returns>The Script class symbol or null if it is not defined.</returns>
+        protected INamedTypeSymbol CommonBindScriptClass()
+        {
+            string scriptClassName = this.Options.ScriptClassName ?? "";
+
+            string[] parts = scriptClassName.Split('.');
+            INamespaceSymbol container = this.SourceModule.GlobalNamespace;
+
+            for (int i = 0; i < parts.Length - 1; i++)
+            {
+                INamespaceSymbol next = container.GetNestedNamespace(parts[i]);
```

Comment:
I don't think so, declaration table always creates top level type declaration for it. See CreateScriptClass method in src\compilers\csharp\portable\declarations\declarationtreebuilder.cs